core/Period/Range.php

```diff
@@ -587,4 +593,18 @@ public function getParentPeriodLabel()
     {
         return null;
     }
+
+    /**
+     * Returns the max allowed end timestamp for a range. If an enddate after this timestamp is provided, Matomo will
+     * automatically lower the end date to the date returned by this method.
+     * The max supported timestamp is always set to end of the current year plus 10 years.
+     *
+     * @return int
+     */
+    public static function getMaxAllowedEndTimestamp(): int
```

Comment:
Does the method need to be public if not used elsewhere?

Reply:
It doesn't and it might not even need to be static. But this new "Limit" is supposed to be something global for ranges. So it imho should be reusable if needed anywhere else.
We could actually even consider to mark this method as `@api`, so it will be shown in the documentation 🤔

Reply:
Hmmm, ok, well if we want to have it in the API scope, then yeah, keep it public. With the static vs dynamic, it seemed to me we prefer dynamic in a lot of places, even just to get a single thing we instantiate a class.. so I could see that used here as well.